typescript/src/index.ts

```diff
@@ -0,0 +1,333 @@
+  });
+
+  if (!searchTokens[0])
+    throw new Error(`Failed to find a token for the object with ID ${object.id}.`);
+  console.log(
+    `Found token with ID ${searchTokens[0].token_id} in collection ${collection}.`
+  );
+
+  return tokens[0].token_id;
+}
+
+async function queryObjects(
+  objectsClient: ObjectsClient,
+  collection: Collection,
+  objects: CustomerResult[],
+  object: CustomerResult,
+  prop: string
+) {
+
+  console.log(`Querying paginated objects in collection ${collection.name}...`);
+  // Query all objects with pagination enabled
+  const allObjects = await objectsClient.listObjects({
+    collection: collection.name,
+    reason,
+    options: ["unsafe"]
+  });
+
+  if (allObjects.results.length === 0)
+    throw new Error("Failed to query objects in the collection.");
+  console.log(
+    `Found the following objects in collection ${collection.name}:${allObjects.results.map(
+      item => `\n\t${JSON.stringify(item)}`
+    )}`
+  );
+  console.log(`Pagination:\n\t${JSON.stringify(allObjects.paging)}`);
+
+  console.log(`\nQuerying for a specific object ssn by ID in collection ${collection.name}...`);
+
+  // Query only one object with a specific prop
+  const objectWithProp = await objectsClient.listObjects({
+    collection: collection.name,
+    reason,
+    ids: [object.id!],
+    props: [prop]
+  });
+
+  if (objectWithProp.results.length === 0)
+    throw new Error(`Failed to query object-${object.id} in the collection with ${prop} prop.`);
+  console.log(
+    `Found the following objects in collection ${collection.name}:${objectWithProp.results.map(
+      item => `\n\t${JSON.stringify(item)}`
+    )}`
+  );
+
+  console.log(
+    `\nQuerying all the details for a specific object by ID in collection ${collection.name}...`
+  );
+
+  // Query only one object with all the props
+  const objectsWithAllProps = await objectsClient.listObjects({
+    collection: collection.name,
+    reason,
+    ids: [object.id!],
+    options: ["unsafe"]
+  });
+  if (objectsWithAllProps.results.length === 0)
+    throw new Error(`Failed to query object-${object.id} in the collection.`);
+
+  console.log(
+    `Found the following objects in collection ${collection.name}:${
+      objectsWithAllProps.results.map(item => `\n\t${JSON.stringify(item)}`)}`
+  );
+
+  // Query only one object with all the props masked
+  const objectsWithMasks = await objectsClient.listObjects({
+    collection: collection.name,
+    reason,
+    ids: [object.id!],
+    props: collection.properties
+      .filter(item => item.name !== "zip_code_us")
+      .map(item => `${item.name}.mask`)
+  });
+  if (objectsWithMasks.results.length === 0)
+    throw new Error(`Failed to query object-${object.id} in the collection.`);
+
+  console.log(
+    `Found the following objects in collection ${collection.name}:${objectsWithMasks.results.map(
+      item => `\n\t${JSON.stringify(item)}`
+    )}`
+  );
+}
+
+async function deleteObject(
```

Comment:
Why does deleteObject has a searchTokenRequest? Same for tokenizeObject above. This is weird

Reply:
Leftover from the original getting started.
I'll rename it to cleanResources.